### PR TITLE
@waku/react docs

### PR DESCRIPTION
**Pull request by LordGhostX**
`@waku/react` docs guide following https://github.com/waku-org/docs.waku.org/pull/82

Preview: https://waku-guide-git-waku-react-docs-logosdao.vercel.app/guides/js-waku/use-waku-react

### CODE REVIEW

**Comment by hackyguru** [comment]
Good idea to add placeholders like 'coming soon' instead of having the link spaces empty? @LordGhostX 

Screenshot 2023-07-15 at 11 11 31 AM

File: docs/guides/js-waku/use-waku-react.md [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+</TabItem>
+</Tabs>
+
+Next, install the `@waku/react` package using your preferred package manager:
+
+<Tabs groupId="package-manager">
+<TabItem value="npm" label="npm">
+
+```shell
+npm install @waku/react @waku/sdk
```
Comment (fryorcraken) [inline docs/guides/js-waku/use-waku-react.md:39]: @weboko What do you think about making it so that you only need to install `@waku/react`?

File: docs/guides/js-waku/use-waku-react.md [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+<TabItem value="yarn" label="Yarn">
+
+```shell
+yarn add @waku/react @waku/sdk
+```
+
+</TabItem>
+</Tabs>
+
+## Create a Relay Node
```
Comment (fryorcraken) [inline docs/guides/js-waku/use-waku-react.md:52]: Let's remove relay node from all docs. It's pointless to use it at this point in time.

File: docs/guides/js-waku/use-waku-react.md [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+
+```js title="App.js"
+import { useCreateLightNode } from "@waku/react";
+import { Protocols } from "@waku/interfaces";
+
+function App() {
+	// Create and start a Light Node and wait for remote peers
+	const { node, error, isLoading } = useCreateLightNode({
+		options: { defaultBootstrap: true },
+		protocols: [Protocols.LightPush, Protocols.Filter],
```
Comment (fryorcraken) [inline docs/guides/js-waku/use-waku-react.md:87]: I don't think we need to specify protocols, if possible let's remove it.

File: docs/guides/js-waku/use-waku-react.md [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+	// Use these to handle states and render the UI in your application
+}
+```
+
+## Create an Encoder and Decoder
+
+Use the `useCreateContentPair()` hook to create a message `encoder` and `decoder` pair:
+
+```js title="App.js"
+import { useCreateContentPair } from "@waku/react";
```
Comment (fryorcraken) [inline docs/guides/js-waku/use-waku-react.md:102]: I thought we removed this API. I think it may be confusing for the reader that content codec are created together (encoder for write, decoder for read).
This is only useful in a very specific case: a public channel where everyone reads and writes.

I think it limits the creativity of the reader but also confuses them if they intend to do any other sort of communication (broadcast, 1:1, etc).

Instead, I'd recommend one API to create the encoder and one for the decoder. As done in `@waku/core`.

Cc @weboko

File: docs/guides/js-waku/use-waku-react.md [hunk 5]
```diff
@@ -0,0 +1,356 @@
...
+		options: { defaultBootstrap: true },
+		protocols: [Protocols.LightPush, Protocols.Filter],
+	});
+
+	// Choose a content topic and create a decoder
+	const contentTopic = "/waku-react-guide/1/message/utf8";
+	const { decoder } = useCreateContentPair(contentTopic);
+
+	// Receive messages from Filter subscription
+	const { error, messages, isLoading } = useFilterMessages({ node, decoder });
```
Comment (fryorcraken) [inline docs/guides/js-waku/use-waku-react.md:188]: What happen if we receive more messages? I'd expect using a callback here. Cc @weboko

File: docs/guides/js-waku/use-waku-react.md [hunk 6]
```diff
@@ -0,0 +1,356 @@
...
+	useCreateContentPair,
+	useStoreMessages,
+} from "@waku/react";
+import { Protocols, PageDirection } from "@waku/interfaces";
+
+function App() {
+	// Create and start a Light Node and wait for remote peers
+	const { node } = useCreateLightNode({
+		options: { defaultBootstrap: true },
+		protocols: [Protocols.LightPush, Protocols.Filter],
```
Comment (fryorcraken) [inline docs/guides/js-waku/use-waku-react.md:214]: Store protocol is missing. We are lucky it works because our fleet has all 3 protocols. I recommend to just not use the `protocol` argument in this guide.

File: docs/guides/js-waku/use-waku-react.md [hunk 7]
```diff
@@ -0,0 +1,356 @@
...
+	});
+
+	// Choose a content topic and create a decoder
+	const contentTopic = "/waku-react-guide/1/message/utf8";
+	const { decoder } = useCreateContentPair(contentTopic);
+
+	// Set the query options
+	const options = {
+		pageDirection: PageDirection.BACKWARD,
+		pageSize: 20,
```
Comment (fryorcraken) [inline docs/guides/js-waku/use-waku-react.md:224]: I believe `pageSize` is optimized for 10 on nwaku side.
@jm-clius can you please confirm? What about for PostgreSQL @Ivansete-status ?

Don't show the usage of this specific option. let the dev use the default options.

File: docs/guides/js-waku/use-waku-react.md [hunk 8]
```diff
@@ -0,0 +1,356 @@
...
+	console.log(messages);
+
+	// "error" captures any error that occurs during message retrieval
+	// "messages" contains a list of messages retrieved from the Store peer
+	// "isLoading" indicates whether the node is still retrieving messages
+	// Use these to handle states and render the UI in your application
+}
+```
+
+:::info
```
Comment (fryorcraken) [inline docs/guides/js-waku/use-waku-react.md:238]: Would be good to document somewhere the optimal usage of Waku store, as set by Lorenzo. @jm-clius can you assist?

File: docs/guides/js-waku/use-waku-react.md [hunk 9]
```diff
@@ -0,0 +1,356 @@
...
+To explore the available query options, please refer to the [Store Query Options](/guides/js-waku/store-retrieve-messages#store-query-options) guide.
+:::
+
+## Using @waku/react Providers
+
+The `@waku/react` package provides a collection of [context providers](https://react.dev/reference/react/createContext#provider) to pass configuration options throughout the component hierarchy:
+
+### `RelayNodeProvider`
```
Comment (fryorcraken) [inline docs/guides/js-waku/use-waku-react.md:246]: let's not document relay nodes.

**Review by fryorcraken — COMMENTED** [review]
tl;dr:
- let's remove protocols
- let's remove relay
- let's not use content pair creator

File: docs/guides/js-waku/use-waku-react.md (hunk 1 above)
Comment (LordGhostX) [reply docs/guides/js-waku/use-waku-react.md:39]: I only used the `utf8ToBytes` function from `@waku/sdk`. I can remove the explicit installation as npm/yarn will install it alongside `@waku/react`

Comment (fryorcraken) [reply docs/guides/js-waku/use-waku-react.md:39]: Nah it's best to explicitely install what you use, some checker will send an error if you don't

File: docs/guides/js-waku/use-waku-react.md [hunk 10]
```diff
@@ -0,0 +1,356 @@
...
+		options: { defaultBootstrap: true },
+		protocols: [Protocols.LightPush, Protocols.Filter],
+	});
+
+	// Choose a content topic and create an encoder
+	const contentTopic = "/waku-react-guide/1/message/utf8";
+	const { encoder  } = useCreateContentPair(contentTopic);
+
+	// Wait for the node to finish loading before sending messages
+	// (isLoading === false)
```
Comment (fryorcraken) [inline docs/guides/js-waku/use-waku-react.md:143]: How would the code look like?

I think this example needs to be a bit more verbose. Where you actually define a button to send messages and the button is `disabled` whne `isLoading` is true.

File: docs/guides/js-waku/use-waku-react.md (hunk 5 above)
Comment (fryorcraken) [reply docs/guides/js-waku/use-waku-react.md:188]: Similar to the "button" comment above.

I think we should have a component where we display messages received.

The final result should be something like https://github.com/waku-org/js-waku-examples/tree/master/examples/relay-reactjs-chat but it uses light push + filter + store.

File: docs/guides/js-waku/use-waku-react.md [hunk 11]
```diff
@@ -0,0 +1,356 @@
...
+	const { error, messages, isLoading } = useFilterMessages({ node, decoder });
+	console.log(messages);
+
+	// "error" captures any error that occurs while receiving messages
+	// "messages" contains a list of messages the subscription received
+	// "isLoading" indicates whether the node is still subscribing to Filter
+	// Use these to handle states and render the UI in your application
+}
+```
+
```
Comment (fryorcraken) [inline docs/guides/js-waku/use-waku-react.md:197]: In the website there is an info block "wait for the messages to finish loading before processing them..".

I am not sure this is very clear. Let's improve the code first to display messages in a component and also have it wait for node to be loaded before it calls `useFilterMessages`.

File: docs/guides/js-waku/use-waku-react.md (hunk 1 above)
Comment (weboko) [reply docs/guides/js-waku/use-waku-react.md:39]: > @weboko What do you think about making it so that you only need to install `@waku/react`?

this would be ideal, but to make it "right" I think there should be a set of abstractions for `React` ready
meaning I would avoid re-exporting everything from `@waku/sdk` through `@waku/react`

File: docs/guides/js-waku/use-waku-react.md (hunk 4 above)
Comment (weboko) [reply docs/guides/js-waku/use-waku-react.md:102]: yes, that's what we agreed on and it is to be addressed in this issue https://github.com/waku-org/js-waku/issues/1060#issuecomment-1447280851 

some of them were closed by the latest release of the `@waku/react` but this one not

File: docs/guides/js-waku/use-waku-react.md (hunk 5 above)
Comment (weboko) [reply docs/guides/js-waku/use-waku-react.md:188]: > What happen if we receive more messages? I'd expect using a callback here. Cc @weboko

it will be updated reactively though subscription [here](https://github.com/waku-org/waku-ui/blob/dce1bb6301d152b320fa408d8bd07e9cfe21dc66/src/useFilterMessages.ts#L61)

File: docs/guides/js-waku/use-waku-react.md [hunk 12]
```diff
@@ -0,0 +1,356 @@
+---
+title: "Build React DApps Using @waku/react"
+---
+
+The [@waku/react](https://www.npmjs.com/package/@waku/react) package provides components and UI adapters to integrate `js-waku` into React applications effortlessly. This guide provides detailed steps for using `@waku/react` in your project.
```
Comment (weboko) [inline docs/guides/js-waku/use-waku-react.md:5]: I think we probably can start mentioning  `@waku/sdk` instead of `js-waku` 

cc @fryorcraken

File: docs/guides/js-waku/use-waku-react.md (hunk 10 above)
Comment (weboko) [reply docs/guides/js-waku/use-waku-react.md:143]: it can be as simple as 

```jsx
if (isLoading) {
  return <Loader />
}
```

or something

File: docs/guides/js-waku/use-waku-react.md (hunk 7 above)
Comment (jm-clius) [reply docs/guides/js-waku/use-waku-react.md:224]: Afaik recommended page size is indeed 20. We do limit the number of content topics per query to 10, mostly to reduce complexity in big databases such as those for Status. This is less relevant for postgresql than sqlite.
